### ISSUE TITLE
Expose additional functionality on the cli deploy command to prevent redundant build steps

### ISSUE THREAD

**Issue by Sahiler** [PrefectHQ/prefect#12365]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar request and didn't find it.
- [X] I searched the Prefect documentation for this feature.

### Prefect Version

2.x

### Describe the current behavior

Currently interacting with the build step on the CLI is not a possibility for a prefect deployment. Ideally being able to expose this functionality allows for more templating options for deployment pipelines, especially redundant builds for docker steps.

### Describe the proposed behavior

Something like 

`prefect deploy -n docker-pipeline build=false`

would offer even more flexibility to the prefect.yaml file when interacting with the cli

### Example Use

Have many flows and only want to deploy ones that have a recent change, without rerunning the build step for each one.

**Comment by Sahiler** [comment]
Could be a potential fix/easier workaround for [9921](https://github.com/PrefectHQ/prefect/issues/9921)

**Comment by desertaxle** [comment]
Thanks for the issue @Sahiler! Just to make sure I understand, is the request to be able to skip the build step when deploying via `prefect deploy`?

**Comment by thaparsahil** [comment]
> is the request to be able to skip the build step when deploying via prefect deploy?
Yes, that was the original request. So we can run the "first" deployment with the build/push steps but then ignore it in all the other ones we are deploying at the same time.

**Comment by thaparsahil** [comment]
We ended up moving the build out of the prefect.yaml for this.
We build first, then push the image and use prefect.yaml only for the deployments.
